api/envoy/config/core/v3/health_check.proto

```diff
@@ -126,6 +126,14 @@ message HealthCheck {
     // <envoy_v3_api_msg_type.matcher.v3.StringMatcher>`. See the :ref:`architecture overview
     // <arch_overview_health_checking_identity>` for more information.
     type.matcher.v3.StringMatcher service_name_matcher = 11;
+
+    // Specifies a list of HTTP response statuses considered retriable. If provided, responses in this range
+    // will count towards the configured :ref:`unhealthy_threshold <envoy_v3_api_field_config.core.v3.HealthCheck.unhealthy_threshold>`.
+    // By default all responses not in :ref:`expected_statuses <envoy_v3_api_field_config.core.v3.HealthCheck.HttpHealthCheck.expected_statuses>`
+    // will result in the host being considered immediately unhealthy. Ranges follow half-open semantics of
+    // :ref:`Int64Range <envoy_v3_api_msg_type.v3.Int64Range>`. The start and end of each range are required.
+    // Only statuses in the range [100, 600) are allowed.
```

Comment:
I'm open to documenting this differently or restricting the range but currently expected statuses supersede retriable statuses. i.e. if 200 is expected and retriable, getting one will just count as a successful health check. I had initially thought this would be simpler than validating against any overlaps.

Reply:
It is indeed simpler. I would suggest at least clarifying this as part of the comments

Reply:
Updated docs.